baseband/vlbi_base/file_info.py

```diff
@@ -0,0 +1,179 @@
+# Licensed under the GPLv3 - see LICENSE
+"""Provide a base class for "info" properties.
+
+Loosely based on astropy.utils.data_info.DataInfo
+"""
+from __future__ import division, unicode_literals, print_function
+
+import astropy.units as u
+from astropy.utils.compat.misc import override__dir__
+
+
+class VLBIInfoBase(object):
+    """Container providing a standardized interface to file information."""
+
+    attr_names = ('format',)
+    """Attributes that the container provides."""
+
+    _parent_attrs = ()
+    _parent = None
+
+    def _collect_info(self):
+        # We link to attributes from the parent rather than just overriding
+        # __getattr__ to allow us to look for changes.
+        for attr in self._parent_attrs:
+            setattr(self, attr, getattr(self._parent, attr))
+        self.missing = {}
+
+    def _up_to_date(self):
+        """Determine whether the information we have stored is up to date."""
+        return all(getattr(self, attr) == getattr(self._parent, attr)
+                   for attr in self._parent_attrs)
```

Comment:
I don't think it is possible, since other properties that are determined may depend on these parent propeties; for instance, think of `Mark4FileReader` where someone updates `decade` after initialization - now `start_time` suddenly becomes available. I'll add a comment to the code to make clear this is why it is done this way.